Source/Scene/Model.js

```diff
@@ -4009,6 +4045,151 @@ define([
+
+            var attributes = {};
+            var decodedAttributeData = result.attributeData;
+            for (var attributeName in decodedAttributeData) {
+                if (decodedAttributeData.hasOwnProperty(attributeName)) {
+                    var attribute = decodedAttributeData[attributeName];
+                    var vertexArray = attribute.array;
+                    var vertexBufferView = addNewVertexBuffer(vertexArray, model, context);
+
+                    var data = attribute.data;
+                    data.bufferView = vertexBufferView;
+
+                    attributes[attributeName] = data;
+                }
+            }
+
+            model._decodedData[primitive.mesh + '.primitive.' + primitive.primitive] = {
+                bufferView : decodedBufferView,
+                attributes : attributes
+            };
+        };
+    }
+
+    function parseDraco(model, context) {
+        if (!defined(model.extensionsRequired['KHR_draco_mesh_compression'])
+            || !defined(model.extensionsUsed['KHR_draco_mesh_compression'])) {
+            return;
+        }
+
+        var loadResources = model._loadResources;
+        if (loadResources.primitivesToDecode.length === 0) {
+            if (loadResources.decoding) {
+                // Done decoding
+                return;
+            }
+
+            loadResources.decoding = true;
+
+            var gltf = model.gltf;
+            ForEach.mesh(gltf, function(mesh, meshId) {
+                ForEach.meshPrimitive(mesh, function(primitive, primitiveId) {
+                    if (!defined(primitive.extensions)) {
+                        return;
+                    }
+
+                    var compressionData = primitive.extensions['KHR_draco_mesh_compression'];
+                    if (!defined(compressionData)) {
+                        return;
+                    }
+
+                    var bufferView = gltf.bufferViews[compressionData.bufferView];
+                    var rawBuffer = gltf.buffers[bufferView.buffer];
+                    var data = rawBuffer.extras._pipeline.source;
+                    data = data.slice(0, data.length);
+
+                    loadResources.primitivesToDecode.enqueue({
+                        mesh : meshId,
+                        primitive : primitiveId,
+                        array : data,
+                        bufferView : bufferView,
+                        compressedAttributes : compressionData.attributes
+                    });
+                });
+            });
+        }
+
+        var decoderTaskProcessor = Model._getDecoderTaskProcessor();
+        var taskData = loadResources.primitivesToDecode.peek();
+        var decodingPromises = [];
+        var promise;
+
+        if (defined(taskData)) {
+            promise = decoderTaskProcessor.scheduleTask(taskData, [taskData.array.buffer]);
+        }
+
+        while (defined(promise)) {
```

Comment:
Would it simplify the code to loop over `loadResources.primitivesToDecode.length`?

With that change the code below may only need to be called in one place.
```
            promise = undefined;
            taskData = loadResources.primitivesToDecode.peek();
            if (defined(taskData)) {
                promise = decoderTaskProcessor.scheduleTask(taskData, [taskData.array.buffer]);
            }
```

Reply:
I updated this to be less redundant, but I'm hesitant to iterate through all of the primitives, as we don't necessarily iterate through all the primitives in the queue each frame. It depends on how many task can be scheduled as well.

Reply:
Ah, good point.